### PR TITLE
[*] Smart Elements - Fix error swallowing on load and clean some useless code

### PR DESCRIPTION
**Pull request by ArnaudValensi**
## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request
- [ ] Write changes made in the CHANGELOG.md
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -171,22 +160,22 @@ exports.init = function (Implementation, dependencies) {
 
           if (opts.configDir) {
             directorySmartImplementation = path.resolve('.', opts.configDir);
+
+            if (!fs.existsSync(directorySmartImplementation)) {
+              logger.error('Your Forest modelsDir option you configured does not seem to be an existing directory.');
+              return P.resolve([]);
```
Comment (arnaudbesnier) [inline index.js]: You don't need to declare a Promise here:
`return [];` is enough.

File: index.js [hunk 2]
```diff
@@ -171,22 +160,22 @@ exports.init = function (Implementation, dependencies) {
...
-                return requireAllModels(Implementation,
-                  directorySmartImplementation, false);
-              }
-            });
+            if (fs.existsSync(directorySmartImplementation)) {
+              return requireAllModels(Implementation, directorySmartImplementation);
+            } else {
+              return P.resolve([]);
+            }
+          }
```
Comment (arnaudbesnier) [inline index.js]: Looks like you duplicated the logic.
the `directorySmartImplementation` value and the error log are the only things that are specific to `configDir` presence.

```javascript
var directorySmartImplementation = path.resolve('.', opts.configDir || '/forest');

if (fs.existsSync(directorySmartImplementation)) {
  return requireAllModels(Implementation, directorySmartImplementation);
} else {
  if (opts.configDir) {
    logger.error('The Forest modelsDir option you configured does not seem to be an existing directory.');
  }
  return [];
}
```
Does the same job, with less code/logic.

Comment (ArnaudValensi) [reply index.js]: `path.resolve('.') + '/forest'` is not the same as `path.resolve('.', '/forest')`

Comment (arnaudbesnier) [reply index.js]: then `path.resolve('.', 'forest')` is equivalent to `path.resolve('.') + '/forest'`

**Pull request merged**